From 2d33110ee18fb59a1d6917d598fc1f4eb3c4cf83 Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Mon, 7 May 2012 16:32:10 -0300
Subject: [PATCH] Switch from bcopy to memmove

The bcopy() function is marked as legacy per POSIX.1-2001 and removed by
POSIX-1.2008 thus shouldn't be used.
So switch over to memmove()

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 mtrace.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/mtrace.c b/mtrace.c
index c982a67..64630bf 100644
--- a/mtrace.c
+++ b/mtrace.c
@@ -556,7 +556,7 @@ int send_recv(u_int32_t dst, int type, int code, int tries, struct resp_buf *sav
 		save->rtime = ((tr.tv_sec + JAN_1970) << 16) +
 		    (tr.tv_usec << 10) / 15625;
 		save->len = len;
-		bcopy((char *)igmp, (char *)&save->igmp, ipdatalen);
+		memmove((char *)&save->igmp, (char *)igmp, ipdatalen);
 	    }
 	    return recvlen;
 	}
@@ -646,7 +646,7 @@ void passive_mode(void)
 	base.rtime = ((tr.tv_sec + JAN_1970) << 16) +
 	    (tr.tv_usec << 10) / 15625;
 	base.len = len;
-	bcopy((char *)igmp, (char *)&base.igmp, ipdatalen);
+	memmove((char *)&base.igmp, (char *)igmp, ipdatalen);
 	/*
 	 * If the user specified which traces to monitor,
 	 * only accept traces that correspond to the
-- 
1.7.3.4

